jmxfetch.py

```diff
@@ -255,14 +264,27 @@ def _start(self, path_to_java, java_run_opts, jmx_checks, command, reporter, too
                 subprocess_args.insert(1, opt)
 
             log.info("Running %s" % " ".join(subprocess_args))
-            jmx_process = subprocess.Popen(subprocess_args, close_fds=True)
+
+            # Launch JMXfetch subprocess
+            jmx_process = subprocess.Popen(
+                subprocess_args,
+                close_fds=not redirect_std_streams,  # set to True instead of False when the streams are redirected for WIN compatibility
+                stdout=subprocess.PIPE if redirect_std_streams else None,
+                stderr=subprocess.PIPE if redirect_std_streams else None
+            )
             self.jmx_process = jmx_process
 
             # Register SIGINT and SIGTERM signal handlers
             self.register_signal_handlers()
 
-            # Wait for JMXFetch to return
-            jmx_process.wait()
+            if redirect_std_streams:
+                # Wait for JMXFetch to return, and write out the stdout and stderr of JMXFetch to sys.stdout and sys.stderr
+                out, err = jmx_process.communicate()
```

Comment:
My bad, it's way easier with Python 2.7, but for `Popen` (Python 2.6...), it's not possible.